src/common/time.rs

```diff
@@ -83,3 +59,36 @@ impl Time {
         }
     }
 }
+
+#[cfg(all(feature = "server", feature = "http1"))]
+macro_rules! check_timer {
+    ($me:ident.$opt:ident, $timer:ident, $closure:expr) => {
+        match $me.$opt {
+            Dur::Default(Some(dur)) => match $me.$timer {
+                Time::Empty => {
+                    warn!(concat!(
+                        "timeout `",
+                        stringify!($opt),
+                        "` has default, but no timer set",
+                    ));
+                }
+                Time::Timer(..) => {
+                    $closure(dur);
+                }
+            },
+            Dur::Configured(Some(dur)) => match $me.$timer {
+                Time::Empty => panic!(concat!(
+                    "timeout `",
+                    stringify!($opt),
+                    "` set, but no timer set",
+                )),
+                Time::Timer(..) => {
+                    $closure(dur);
+                }
+            },
+            Dur::Default(None) | Dur::Configured(None) => {
+                // do nothing
+            }
+        }
+    };
+}
```

Comment:
I went back and forth with it being a method of `Time`. The reason I liked a macro was because I wanted to log the option name, and passing a string of the name like `self.timer.check(self.header_read_timeout, "header_read_timeout")` seemed unfortunate. Perhaps it's not as big a deal, I could be convinced otherwise.

Reply:
I understand the reason for using a macro. In my preference, the readability difficulty which is added by using a macro seems not worth only the cleaner interface for logging at this case. Although I agree that the log setting does not seem look happy, I'm inclined to avoid macro considering the readability of macro.

Reply:
That's fair, thanks for convincing me! I'll change it to a method.